util/rowcodec/rowcodec_test.go

```diff
@@ -82,172 +76,183 @@ func (s *testSuite) TestEncodeLargeSmallReuseBug(c *C) {
-			c.Assert(d, DeepEquals, t.dt)
-		}
-
-		// decode to chunk.
-		cDecoder := rowcodec.NewChunkDecoder(cols, []int64{-1}, nil, sc.TimeZone)
-		chk := chunk.New(fts, 1, 1)
-		err = cDecoder.DecodeToChunk(newRow, kv.IntHandle(handleValue), chk)
-		c.Assert(err, IsNil)
-		chkRow := chk.GetRow(0)
-		cdt := chkRow.GetDatumRow(fts)
-		for i, t := range testData {
-			d := cdt[i]
-			if d.Kind() == types.KindMysqlDecimal {
-				c.Assert(d.GetMysqlDecimal(), DeepEquals, t.bt.GetMysqlDecimal())
-			} else {
-				c.Assert(d, DeepEquals, t.bt)
-			}
-		}
-
-		// decode to old row bytes.
-		colOffset := make(map[int64]int)
-		for i, t := range testData {
-			colOffset[t.id] = i
-		}
-		bDecoder := rowcodec.NewByteDecoder(cols, []int64{-1}, nil, nil)
-		oldRow, err := bDecoder.DecodeToBytes(colOffset, kv.IntHandle(handleValue), newRow, nil)
-		c.Assert(err, IsNil)
-		for i, t := range testData {
-			remain, d, err := codec.DecodeOne(oldRow[i])
-			c.Assert(err, IsNil)
-			c.Assert(len(remain), Equals, 0)
-			if d.Kind() == types.KindMysqlDecimal {
-				c.Assert(d.GetMysqlDecimal(), DeepEquals, t.bt.GetMysqlDecimal())
-			} else {
-				c.Assert(d, DeepEquals, t.bt)
-			}
-		}
-	}
-
-	// encode & decode signed int.
-	testDataSigned := []testData{
+	tests := []struct {
+		name     string
+		testData []testData
+	}{
 		{
-			handleID,
-			types.NewFieldType(mysql.TypeLonglong),
-			types.NewIntDatum(handleValue),
-			types.NewIntDatum(handleValue),
-			nil,
-			true,
+			"signed int",
+			[]testData{
+				{
+					handleID,
+					types.NewFieldType(mysql.TypeLonglong),
+					types.NewIntDatum(handleValue),
+					types.NewIntDatum(handleValue),
+					nil,
+					true,
+				},
+				{
+					10,
+					types.NewFieldType(mysql.TypeLonglong),
+					types.NewIntDatum(1),
+					types.NewIntDatum(1),
+					nil,
+					false,
+				},
+			},
 		},
 		{
-			10,
-			types.NewFieldType(mysql.TypeLonglong),
-			types.NewIntDatum(1),
-			types.NewIntDatum(1),
-			nil,
-			false,
+			"unsigned int",
+			[]testData{
+				{
+					handleID,
+					withUnsigned(types.NewFieldType(mysql.TypeLonglong)),
+					types.NewUintDatum(uint64(handleValue)),
+					types.NewUintDatum(uint64(handleValue)), // decode as bytes will uint if unsigned.
+					nil,
+					true,
+				},
+				{
+					10,
+					types.NewFieldType(mysql.TypeLonglong),
+					types.NewIntDatum(1),
+					types.NewIntDatum(1),
+					nil,
+					false,
+				},
+			},
 		},
 	}
-	encodeAndDecodeHandle(c, testDataSigned)
 
-	// encode & decode unsigned int.
-	testDataUnsigned := []testData{
-		{
-			handleID,
-			withUnsigned(types.NewFieldType(mysql.TypeLonglong)),
-			types.NewUintDatum(uint64(handleValue)),
-			types.NewUintDatum(uint64(handleValue)), // decode as bytes will uint if unsigned.
-			nil,
-			true,
-		},
-		{
-			10,
-			types.NewFieldType(mysql.TypeLonglong),
-			types.NewIntDatum(1),
-			types.NewIntDatum(1),
-			nil,
-			false,
-		},
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
```

Comment:
the body of `t.Run` is from encodeAndDecodeHandle with trivial rename or gocheck to testify.

Reply:
ditto others